templates/tapestry-d3-vue/src/components/RootNodeButton.vue

```diff
@@ -0,0 +1,52 @@
+<template>
+  <div id="root-node-container">
+    <div id="root-node-btn" @click="showModal">
+      <i class="fas fa-plus-circle fa-5x"></i>
+      <div id="root-node-label">Add Root Node</div>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'root-node-button',
+  data() {
+    return {}
+  },
+  methods: {
+    showModal() {
+      console.log('here')
+      // Populate title
+      $('#createNewNodeModalLabel').text("Add root node")
+      $("#submit-add-new-node").hide()
```

Comment:
Please do not use this hide/show logic as it's very long and complicated. Bind your elements to models and then they should automatically be filled in / updated.

Reply:
Could we please leave this for later battle when we refactor to get rid of all the jQuery scripts? Ideally, I think there should be one single button that does all the edit/add rootNode/nodes in the modal.

Reply:
What is the point of using VueJS if we don't use models? It's very weird for me to not use models as they are very easy to use and make life a whole lot simpler. I would really prefer we use v-model instead of manually having to fill in fields like this (unless you are out of time, in which case I'll let you prioritize)